querier.go

```diff
@@ -791,31 +794,50 @@ func (it *chunkSeriesIterator) Seek(t int64) (ok bool) {
 		t = it.mint
 	}
 
-	for ; it.chunks[it.i].MaxTime < t; it.i++ {
-		if it.i == len(it.chunks)-1 {
-			return false
-		}
+	// Return early if already past t.
+	if t0, _ := it.cur.At(); t0 >= t {
```

Comment:
Just realized that this might call `At()` on an iterator that `Next()` or `Seek()` might not have been called on. It works if one assumes all timestamps to be strictly positive and that the iterator will return a timestamp of 0 when uninitialized.

Should I remove it?

Reply:
Well, crap. This is the thing that makes `Seek()` not advance the iterator when seeking for a timestamp that's in the past (compared to the iterator's position).

So it looks to me like we need to change the contract of either `chunkenc.Iterator.At()` to return something like `math.MinInt64`; or that of `tsdb.SeriesIterator.Seek()` to always advance the iterator. The latter seems more radical.

Reply:
I've changed `chunkenc.Iterator.At()` to always return `math.MinInt64` before it has advanced (and documented it as such). Everything is good in the world again.

Reply:
Yep, before the contract used to be that `At()` shouldn't be called before `Next()` or `Seek()` and it shouldn't be called after they return `false`.

I like the newer contract of returning `math.MinInt64` for non-initialised `Iterator`s.